data-prepper-plugins/otel-trace-raw-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/oteltrace/OTelTraceRawPrepper.java

```diff
@@ -6,59 +6,73 @@
 import com.amazon.dataprepper.model.prepper.AbstractPrepper;
 import com.amazon.dataprepper.model.record.Record;
 import com.amazon.dataprepper.plugins.prepper.oteltrace.model.OTelProtoHelper;
+import com.amazon.dataprepper.plugins.prepper.oteltrace.model.RawSpan;
 import com.amazon.dataprepper.plugins.prepper.oteltrace.model.RawSpanBuilder;
+import com.amazon.dataprepper.plugins.prepper.oteltrace.model.RawSpanSet;
+import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.primitives.Ints;
 import io.micrometer.core.instrument.Counter;
-
 import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
 import io.opentelemetry.proto.trace.v1.InstrumentationLibrarySpans;
 import io.opentelemetry.proto.trace.v1.ResourceSpans;
 import io.opentelemetry.proto.trace.v1.Span;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.math.BigDecimal;
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.DelayQueue;
+import java.util.concurrent.Delayed;
+import java.util.concurrent.TimeUnit;
 
 
 @DataPrepperPlugin(name = "otel_trace_raw_prepper", type = PluginType.PREPPER)
 public class OTelTraceRawPrepper extends AbstractPrepper<Record<ExportTraceServiceRequest>, Record<String>> {
 
-    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
-    private static final String INSTRUMENTATION_LIBRARY_SPANS = "instrumentationLibrarySpans";
-    private static final String INSTRUMENTATION_LIBRARY = "instrumentationLibrary";
-    private static final String SPANS = "spans";
-    private static final String RESOURCE = "resource";
-    private static final String ATTRIBUTES = "attributes";
-    private static final String START_TIME_UNIX_NANOS = "startTimeUnixNano";
-    private static final String END_TIME_UNIX_NANOS = "endTimeUnixNano";
-    private static final String START_TIME = "startTime";
-    private static final String END_TIME = "endTime";
-    private static final BigDecimal MILLIS_TO_NANOS = new BigDecimal(1_000_000);
-    private static final BigDecimal SEC_TO_MILLIS = new BigDecimal(1_000);
+    private static final long SEC_TO_MILLIS = 1_000L;
     private static final Logger log = LoggerFactory.getLogger(OTelTraceRawPrepper.class);
 
     public static final String SPAN_PROCESSING_ERRORS = "spanProcessingErrors";
     public static final String RESOURCE_SPANS_PROCESSING_ERRORS = "resourceSpansProcessingErrors";
     public static final String TOTAL_PROCESSING_ERRORS = "totalProcessingErrors";
 
+    private final long gcInterval;
```

Comment:
Maybe rename the garbage collection concept to something like orphan flush interval to make it more obvious what we mean by "garbage collection"

Reply:
Agreed. "Garbage collection" can be misleading as those spans are not garbage.